### PR TITLE
Java 14-16 compatability

### LINKED ISSUE #10113: Using Gson to marshal many java.* classes is platform specific
Gson reflects into private fields of types it does not have [an explicit understanding](https://github.com/google/gson/blob/f649e051411e092f0123878e16c5132500a2d01e/gson/src/main/java/com/google/gson/Gson.java#L231-L276) of, which means it depends on the vendor/version of the JVM how/if the type is serialized. This will also fail in java >= 16 due to illegal access being denied due to JEP 396.

The solution to this would be to create TypeAdapters for any types Gson does not understand already, or not use the type.

Some (__not all__) cases of this:
 - `java.util.UUID` used all over in the session stuff
 - `java.awt.Color` used in screen markers (this is painful because its in the config and has an *interesting* format)

### PR DESCRIPTION
**Pull request by abextm**
Includes workarounds for:
 - [JDK-8173978](https://bugs.openjdk.java.net/browse/JDK-8173978) (Java 14)
 - [JEP396/JDK-8255363](https://bugs.openjdk.java.net/browse/JDK-8255363) (Java 16)

Closes #10113

### CODE REVIEW

File: http-api/src/main/java/net/runelite/http/api/gson/ColorTypeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	{
+		switch (in.peek())
+		{
+			case NULL:
+				in.nextNull();
+				return null;
+			case BEGIN_OBJECT:
+				in.beginObject();
+				double value = 0;
+				for (; in.peek() != JsonToken.END_OBJECT; )
```
Comment (Nightfirecat) [inline http-api/src/main/java/net/runelite/http/api/gson/ColorTypeAdapter.java:57]:
```suggestion
				while (in.peek() != JsonToken.END_OBJECT)
```
for all the type adapter loops

File: http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+package net.runelite.http.api.gson;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+import java.time.Instant;
+
+// Just add water!
```
Comment (Nightfirecat) [inline http-api/src/main/java/net/runelite/http/api/gson/InstantTypeAdapter.java:34]: :joy:

File: http-api/src/main/java/net/runelite/http/api/gson/ColorTypeAdapter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import com.google.gson.stream.JsonWriter;
+import java.awt.Color;
+import java.io.IOException;
+
+public class ColorTypeAdapter extends TypeAdapter<Color>
+{
+	@Override
+	public void write(JsonWriter out, Color value) throws IOException
+	{
+		int rgba = value.getRGB();
```
Comment (Nightfirecat) [inline http-api/src/main/java/net/runelite/http/api/gson/ColorTypeAdapter.java:39]: Do we not need to handle null color values for this?

**Review by Nightfirecat — COMMENTED** [review]
I can't speak to the ReflectUtil change, but e9b3c16 looks good to me

**Review by Adam- — COMMENTED** [review]
The only gson instance left being created I can see is in the skill calc, did you intentionally keep that? I don't think it requires using any of the new serializers though so maybe it's ok.

**Pull request merged**